### PR TITLE
Replace UI checks in firefox with web scraping

### PR DESCRIPTION
**Pull request by asmorodskyi**

### CODE REVIEW

**Comment by asmorodskyi** [comment]
@mslacken

**Comment by asmorodskyi** [comment]
after implementing https://progress.opensuse.org/issues/34174 , there is no image for HPC with gnome . so there is no way to check ganglia web UI with firefox. which force us to use web scraping instead

**Comment by asmorodskyi** [comment]
http://kimball.arch.suse.de/tests/1181
http://kimball.arch.suse.de/tests/1182
http://kimball.arch.suse.de/tests/1183

proof runs

File: tests/hpc/ganglia_server.pm [hunk 1]
```diff
@@ -50,20 +50,25 @@ sub run {
...
-    assert_screen('ganglia-node-report', 60);
+    assert_script_run("curl -s \"$image_url\" > test2.png");
+    assert_script_run('file test2.png | grep -q "PNG image data"');
 
     # tell client that server is done
     barrier_wait('GANGLIA_SERVER_DONE');
+    # barrier check period is 5 seconds , we sleeping for 6 to cover
+    # case when ganglia-client already reach this barrier but currently
+    # in sleep so it will die with owner already leave error on next check
+    sleep 6;
```
Comment (czerw) [inline tests/hpc/ganglia_server.pm]: This is  workaround at the moment. Final solution will be covered in https://progress.opensuse.org/issues/35218 .

File: lib/serial_terminal.pm [hunk 2]
```diff
@@ -105,7 +105,7 @@ sub select_virtio_console {
     }
 
     bmwqemu::fctwarn("Cannot use virtio console on s390 on ZKVM") if $is_virtio;
-    select_console('root-console');
+    select_console('root-console', ensure_tty_selected => 1);
```
Comment (czerw) [inline lib/serial_terminal.pm]: Shouldn't be better to split it into separate commit/pr? It is not mentioned in the description and  I don't see relation to ganglia test.

**Comment by asmorodskyi** [comment]
@czerw  - done

**Comment by czerw** [comment]
lgtm

**Comment by asmorodskyi** [comment]
merging based on Petr comment
